app/src/main/java/dev/arkbuilders/arkmemo/data/repositories/TextNotesRepo.kt

```diff
@@ -1,13 +1,114 @@
 package dev.arkbuilders.arkmemo.data.repositories
 
+import android.util.Log
+import dev.arkbuilders.arklib.computeId
+import dev.arkbuilders.arklib.user.properties.PropertiesStorageRepo
+import dev.arkbuilders.arkmemo.data.ResourceMeta
+import dev.arkbuilders.arkmemo.di.IO_DISPATCHER
+import dev.arkbuilders.arkmemo.di.PropertiesStorageModule.STORAGE_SCOPE
+import dev.arkbuilders.arkmemo.models.Content
+import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.withContext
 import dev.arkbuilders.arkmemo.models.TextNote
-import java.nio.file.Path
+import dev.arkbuilders.arkmemo.preferences.MemoPreferences
+import java.nio.file.Files
+import javax.inject.Inject
+import javax.inject.Named
+import javax.inject.Singleton
+import kotlin.io.path.extension
+import kotlin.io.path.fileSize
+import kotlin.io.path.forEachLine
+import kotlin.io.path.getLastModifiedTime
+import kotlin.io.path.name
+import kotlin.io.path.writeLines
 
-interface TextNotesRepo {
-    suspend fun init(root: Path, scope: CoroutineScope)
-    suspend fun save(note: TextNote)
-    suspend fun delete(note: TextNote): Int
-    suspend fun read(): List<TextNote>
+@Singleton
+class TextNotesRepo @Inject constructor(
```

Comment:
We provided this in the DI so no need @Singleton here. Also this should extends from interface, not implementation

Reply:
I forgot :)